### PR TITLE
Add a setting for always closing drawers when using home button

### LINKED ISSUE #57: Bug with navbar. It does not go to the desktop when I hit the "home" button
I have noticed this problem. I am using android 13 with navbar on screen. I use the latest version (F-Droid)

### PR DESCRIPTION
**Pull request by esensar**
This closes #57

### CODE REVIEW

**Comment by esensar** [comment]
This is a continuation of #85

**Comment by tibbi** [comment]
this doesnt work at all on my Android 13, nothing happens at pressing Home

**Comment by esensar** [comment]
How have you tested it? The thing is, we can't really detect home button press, so I just made sure to clean up open drawers when the activity has been moved to the background. This means it will not work when pressing home button from the drawer, but only when in other apps.

**Comment by tibbi** [comment]
well, I pressed Home and nothing at all happened.

File: app/src/main/res/values-bg/strings.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <string name="hidden_icons_placeholder">Някои приложения не могат да бъдат деинсталирани поради системни ограничения, но можете поне да скриете иконите им, за да не ги виждате.</string>
     <string name="app_drawer_settings">App drawer</string>
     <string name="close_app_drawer_on_app_open">Close app drawer on opening an app</string>
+    <string name="close_app_drawer_on_home">Close app drawer when home button is used</string>
```
Comment (tibbi) [inline app/src/main/res/values-bg/strings.xml:16]: change it to "Close app drawer on pressing Home"

**Comment by tibbi** [comment]
this behaviour just feels weird :/ Lets make it always work like that, without the setting. I misunderstood the setting at the previous PR. However, currently if you are at Launcher settings or have app drawer open and press Home, nothing happens.

**Comment by tibbi** [comment]
and if I open an app and press Home, the home screen icon appearing animation is always played. Like the screen was always recreated instead of just resumed.

**Comment by esensar** [comment]
It should behave better now. There was no need to do any `onResume` hacks, `onNewIntent` works as expected (it receives the `HOME` intent properly now), I just needed to update target activity for that intent.

**Comment by tibbi** [comment]
well, the animation is still there though on my Samsung. You cant reproduce it on any device?

**Comment by esensar** [comment]
Can you record a video? I don't see any extra animation on my devices...

https://github.com/SimpleMobileTools/Simple-Launcher/assets/2764831/379d6cd3-4591-4144-87ad-8d6a2c95bd64


https://github.com/SimpleMobileTools/Simple-Launcher/assets/2764831/92efec79-a42a-4fa9-9888-f75557a669c2

**Comment by tibbi** [comment]
you literally have an appearing animation on the pixel on the first second, as you press Home while the drawer is open

**Comment by esensar** [comment]
> you literally have an appearing animation on the pixel on the first second, as you press Home while the drawer is open

Yeah, I see what you are talking about, but it is not controlled by the app
I will see if there is a way to suppress that animation

**Comment by tibbi** [comment]
well, it isnt there in Nova and other downloaded launchers, so it has to be doable

**Comment by esensar** [comment]
So, I have made a small change that should make it similar to Nova. Looking at Nova, some animation is still there, but it is concealed with a different animation (the drawer sliding down).


https://github.com/SimpleMobileTools/Simple-Launcher/assets/2764831/5831ba46-2d52-48af-ae91-8f7467cf5a81


https://github.com/SimpleMobileTools/Simple-Launcher/assets/2764831/91edd328-a50a-4bd4-bbf5-eae6c364e68f

**Comment by tibbi** [comment]
it didnt solve it for me, but Ill accept it to move forward. The app still needs many improvements, lets go step by step. Thanks

**Pull request merged**